### DISCUSSION TITLE
60 seconds delay observed for runOnce events scheduled at same time

### DISCUSSION THREAD

**Discussion by ESHA1998** [quartznet/quartznet#1147] [General] [answered]
Hello everyone

We are using quartz-scheduler for scheduling around 500 jobs of Run-once type scheduled for the same time. We are observing the following behaviour-

a) Although there are 10 threads present in quartz scheduler(org.quartz.threadPool.threadCount), only a single worker thread executes sequentially at a particular instance of time eg- if a worker thread-1 is performing a task of processing an event, it would complete it first and then the other thread would start with the same.(This behaviour is observed when we are not using @DisallowConcurrentExecution annotation)

b) Dividing the time frame of a minute, we get the following observations-
i)Time taken by a worker thread to complete its processing – 20 ms approx.
ii)Time taken for a switch between two threads ie one thread completing its execution and other thread starting it – 150-200 ms approx.
iii) Number of events processed on an avg in a sec = 5
iv) No of events expected to be processed in a min = 5*60=300 

But the actual events processed in a minute on an average remains to be 5. That is further no work is performed in that minute and this acts as a bottleneck towards the performance. Attached are the logs depicting this behaviour.

Parameters used:
1. clusterCheckinInterval - 120000
2. isClustered - true
3. misfireThreshold - 60000
4. useProperties - false
5. instanceId - AUTO
6. threadCount -10 
7. threadPriority - 5

Versions used - 
1. quartz-scheduler - 2.3.2
2. spring-boot - 1.2.4
3. spring-context-support - 5.2.1

What we tried:
1. Reducing the misfireThreshold to 20000
2. Setting org.quartz.scheduler.batchTriggerAcquisitionMaxCount property to 10
But 60 sec delay is still observed.

Please help us in identifying why this behaviour could be persisting and how can this issue be resolved.

[logs1.txt](https://github.com/quartznet/quartznet/files/6236322/logs1.txt)

**Comment by lahma** [comment, marked as answer]
Sorry but this repo is for the .NET version 🙂

**Reply by ESHA1998** [reply]
Haha, yeah moved it to quartz-scheduler repo. Thanks!
